### PR TITLE
Request historical messages in batches

### PR DESCRIPTION
**Pull request by adambabik**
Sending envelopes one-by-one is inefficient. This PR introduces a flag which can be used to turn on batch sending for historical messages.

Important changes:
- [x] introduced a new way of passing payload to mail server: as a struct serialized with RLP,
- [x] added `Batch` flag to `shhext_requestMessages` params.

Todo:
- [x] add unit tests to verify that a default behaviour is sending envelopes one-by-one,
- [x] validate that new serialization of params works,
- [x] validate that `Batch` set to `true` is working as expected.

This should be merged after #1244.

### CODE REVIEW

File: mailserver/mailserver.go [hunk 1]
```diff
@@ -247,25 +250,66 @@ func (s *WMailServer) DeliverMail(peer *whisper.Peer, request *whisper.Envelope)
...
+		cursor,
+		batch)
+	if err != nil {
+		processRequestErrorsCounter.Inc(1)
+		log.Error("Error while processing mail server request", "err", err)
+		return
+	}
+
+	log.Debug("Processed request successfully")
+	log.Debug("Sending historic message response", "last", lastEnvelopeHash, "next", nextPageCursor)
```
Comment (dshulyak) [inline mailserver/mailserver.go:309]: two logs are intentional?

File: services/shhext/api.go [hunk 2]
```diff
@@ -461,33 +488,23 @@ func makeEnvelop(
...
+		cursor = nil
+	}
+
+	payload := MessagesRequestPayload{
+		Lower:  r.From,
+		Upper:  r.To,
+		Bloom:  createBloomFilter(r),
+		Limit:  r.Limit,
+		Cursor: cursor,
+		Batch:  r.Batch,
```
Comment (dshulyak) [inline services/shhext/api.go]: are you sure that it is necessary for a client to know that envelopes are fetched in batches? this is more of a transport layer detail

**Review by dshulyak — APPROVED** [review]

File: services/shhext/api.go (hunk 2 above)
Comment (adambabik) [reply services/shhext/api.go]: Unfortunately yes. It's the client code that supports or does not support receiving envelopes in batches. The other option was to introduce a method as you mentioned in the other PR.

We will be able to get rid of this property after two releases.

Also, I guess we can hide it from the client... We can set it to `true` by default in this PR.

File: mailserver/mailserver.go (hunk 1 above)
Comment (adambabik) [reply mailserver/mailserver.go:309]: Yes but the first one is not needed. I will remove it.

File: services/shhext/api.go (hunk 2 above)
Comment (dshulyak) [reply services/shhext/api.go]: i meant that it doesn't need to come from r.Batch, can be hardcoded

Comment (adambabik) [reply services/shhext/api.go]: Yes, I changed it exactly this way.

**Review by gravityblast — APPROVED** [review]
nice job! using rlp to encode the request is much better!

LGTM

**Pull request merged**